### PR TITLE
CORPORATION: adding prevState and changed state display in the warehouse UI

### PR DESCRIPTION
**Pull request by Caldwell-74**
based on #857

Corp State now has getters to access nextState and prevState
prep work to include prevState in getCorporation() didnt include that yet 
because an API break is needed to change .state to .nextState to not cause confusion when .prevState is added

also changed how states are displayed in the warehouse ui to clear up the state system

image

### CODE REVIEW

File: src/Corporation/CorporationState.ts [hunk 1]
```diff
@@ -9,20 +9,21 @@ export class CorporationState {
   // Get the name of the current state
   // NOTE: This does NOT return the number stored in the 'state' property,
   // which is just an index for the array of all possible Corporation States.
-  getState(): CorpStateName {
+  get nextState(): CorpStateName {
```
Comment (jjclark1982) [inline src/Corporation/CorporationState.ts]: maybe call these `state.nextName` and `state.prevName`?

Comment (Caldwell-74) [reply src/Corporation/CorporationState.ts]: what do you think about 
.nextStateName
.prevStateName ?

Comment (Caldwell-74) [reply src/Corporation/CorporationState.ts]: oh nvm the state is already in there yeah .nextName .prevName sound good

**Comment by Caldwell-74** [comment]
there was the idea from myhf to use a different progress bar like
image
it would be a bit more complicated because the state name would need to move to "wrap" around correctly but if its wanted i can try to implement it

**Comment by Snarling** [comment]
Looks good! I just had to do a quick commit 0524112 to fix a conflict from #845

**Comment by Caldwell-74** [comment]
just for the future i run "npm run doc" by default
if i catch doc changes outside of the scope of my pr do you want me to make a new pr for those or just put them into that pr like with this?

**Comment by Snarling** [comment]
If you already have doc changes in yours, just include all the fixes. If you don't have your own doc changes, any of the three here are acceptable:

1. Include the doc updates in your existing pr
2. Don't include them, make a separate pr
3. Don't include them, no separate pr

Ideally if any got missed, they'll be caught on someone's PR or on the next changelog update.

**Pull request merged**